include/crow/websocket.h

```diff
@@ -656,6 +637,50 @@ namespace crow
                     delete this;
             }
 
+
+            struct SendMessageType
+            {
+                std::string payload;
+                Connection* self;
+                int opcode;
+
+                void operator()()
+                {
+                    self->send_data_impl(this);
+                }
+
+                SendMessageType() noexcept = default;
+
+                SendMessageType(SendMessageType const&) = delete;
+                SendMessageType& operator=(SendMessageType const&) = delete;
+
+                SendMessageType(SendMessageType&&) noexcept = default;
+                SendMessageType& operator=(SendMessageType&&) noexcept = default;
+            };
+
+            static_assert(
+              std::is_nothrow_move_assignable<SendMessageType>::value &&
+                std::is_nothrow_move_constructible<SendMessageType>::value,
+              "SendMessageType must be nothrow movable!");
+
+            void send_data_impl(SendMessageType* s)
+            {
+                auto header = build_header(s->opcode, s->payload.size());
+                write_buffers_.emplace_back(std::move(header));
+                write_buffers_.emplace_back(std::move(s->payload));
+                do_write();
+            }
+
+            void send_data(int opcode, std::string&& msg)
+            {
+                SendMessageType event_arg{
+                  std::move(msg),
+                  this,
+                  opcode};
+
+                post(std::move(event_arg));
+            }
+
```

Comment:
Oh, 😨 This looks like a total overkill... I'll accept it anyway, but please note more code = more maintenance

Reply:
Okay, I have applied changes for removing lambda capture. Since close() routine does not share any routine with other 4(send_text, binary, ping, pong)  and much less frequently called, I just roll-backed close() function's refactor.

Additionally, I removed unnecessary assertions & operator declartions for `SendMessageType` struct on commit 02c0f8db129f0042da3de4ded3e8b3b6f15927c3